### PR TITLE
Remove `queryKey: false` from docs.

### PR DESCRIPTION
**Pull request by vslinko**

### CODE REVIEW

**Comment by taion** [comment]
Thanks!

File: docs/HashHistoryCaveats.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
...
 
-HTML5 gives us the `pushState` method and the `popstate` event, but in older browsers the only thing we have is the URL. So, when using hash history, you'll see an extra item in your query string that looks something like `_k=123abc`. This is a key that `history` uses to look up persistent state data in `window.sessionStorage` between page loads. If you prefer to use a different query parameter, or to opt-out of this behavior entirely, use the `queryKey` configuration option.
+HTML5 gives us the `pushState` method and the `popstate` event, but in older browsers the only thing we have is the URL. So, when using hash history, you'll see an extra item in your query string that looks something like `_k=123abc`. This is a key that `history` uses to look up persistent state data in `window.sessionStorage` between page loads. If you prefer to use a different query parameter, use the `queryKey` configuration option.
```
Comment (taion) [inline docs/HashHistoryCaveats.md]: Can you change

> when using hash history

to

> when using `location.state` with hash history

or something to that effect?

Comment (vslinko) [inline docs/HashHistoryCaveats.md]: Done.

**Comment by taion** [comment]
:+1:

**Comment by basarat** [comment]
Curious why? 

Refs : https://github.com/reactjs/react-router/issues/1967#issuecomment-233828095 :rose:

**Comment by johnnyreilly** [comment]
Yup - would also love to know; this is useful information!

**Pull request merged**